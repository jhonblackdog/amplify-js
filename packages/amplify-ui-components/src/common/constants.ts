// Namespace
export const AMPLIFY_UI_PREFIX = 'amplify-ui';

// Classnames
export const AMPLIFY_UI_TEXT_INPUT = `${AMPLIFY_UI_PREFIX}-text-input`;
export const AMPLIFY_UI_SCENE = `${AMPLIFY_UI_PREFIX}-scene`;
export const AMPLIFY_UI_SCENE_LOADING = `${AMPLIFY_UI_PREFIX}-scene-loading`;

// Dictionaries

/*
  The below constants will be moved to an i18n solution.
  Storing here to have a common place to pull from once we pick up that work.
*/
// Username common constants
export const USERNAME_SUFFIX = 'username';
export const USERNAME_LABEL = 'Username *';
export const USERNAME_PLACEHOLDER = 'Enter your username';

// Email common constants
export const EMAIL_SUFFIX = 'email';
export const EMAIL_LABEL = 'Email Address *';
export const EMAIL_PLACEHOLDER = 'amplify@example.com';

// Code common constants
export const CODE_SUFFIX = 'code';
export const CODE_LABEL = 'Verification code';
export const CODE_PLACEHOLDER = 'Enter code';
export const SEND_CODE = 'Send Code';
export const CONFIRM = 'Confirm';

// Phone common constants
export const PHONE_SUFFIX = 'phone';
export const PHONE_LABEL = 'Phone Number *';
export const PHONE_PLACEHOLDER = '(555) 555-1212';

// Password common constants
export const FORGOT_PASSWORD_TEXT = 'Forgot your password?';
export const RESET_PASSWORD_TEXT = 'Reset password';
export const RESET_YOUR_PASSWORD = 'Reset your password';
export const PASSWORD_SUFFIX = 'password';
export const PASSWORD_LABEL = 'Password *';
export const PASSWORD_PLACEHOLDER = 'Enter your password';
export const BACK_TO_SIGN_IN = 'Back to Sign In';

// Country Dial Code common constants
export const COUNTRY_DIAL_CODE_SUFFIX = 'country-dial-code-select';
export const COUNTRY_DIAL_CODE_DEFAULT = '+1';

// Sign In common constants
// TODO: Change to SIGN_IN_HEADER_TEXT
export const HEADER_TEXT = 'Sign into your account';
// TODO: Change to SIGN_IN_SUBMIT_BUTTON_TEXT
export const SUBMIT_BUTTON_TEXT = 'Sign in';
export const CREATE_ACCOUNT_TEXT = 'Create account';
export const NO_ACCOUNT_TEXT = 'No account?';
export const CONFIRM_SMS_CODE = 'Confirm SMS Code';
export const CONFIRM_TOTP_CODE = 'Confirm TOTP Code';
export const SIGN_IN_WITH_AMAZON = 'Sign In with Amazon';
export const SIGN_IN_WITH_AUTH0 = 'Sign In with Auth0';
export const SIGN_IN_WITH_FACEBOOK = 'Sign In with Facebook';
export const SIGN_IN_WITH_GOOGLE = 'Sign In with Google';
export const SIGN_IN_WITH_AWS = 'Sign in with AWS';

// Sign Up common constants
export const SIGN_UP_HEADER_TEXT = 'Create a new account';
export const SIGN_UP_SUBMIT_BUTTON_TEXT = 'Create account';
export const HAVE_ACCOUNT_TEXT = 'Have an account?';
export const SIGN_IN_TEXT = 'Sign in';
export const SIGN_UP_USERNAME_PLACEHOLDER = 'Create a username';
export const SIGN_UP_PASSWORD_PLACEHOLDER = 'Create a password';

// Confirm Sign Up common constants
export const CONFIRM_SIGN_UP_HEADER_TEXT = 'Confirm Sign up';
export const CONFIRM_SIGN_UP_SUBMIT_BUTTON_TEXT = 'Confirm';
export const CONFIRM_SIGN_UP_CODE_LABEL = 'Confirmation Code';
export const CONFIRM_SIGN_UP_CODE_PLACEHOLDER = 'Enter your code';
export const CONFIRM_SIGN_UP_LOST_CODE = 'Lost your code?';
export const CONFIRM_SIGN_UP_RESEND_CODE = 'Resend Code';

// Sign Out
export const SIGN_OUT = 'Sign Out';

// Auth Keys
export const AUTH_SOURCE_KEY = 'amplify-auth-source';

// Error message Common Constants
export const PHONE_EMPTY_ERROR_MESSAGE = 'Phone number can not be empty';
export const NO_AUTH_MODULE_FOUND = 'No Auth module found, please ensure @aws-amplify/auth is imported';
