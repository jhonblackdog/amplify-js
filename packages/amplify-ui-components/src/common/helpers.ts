import { cx } from 'emotion';

// TODO: rename
export const styleNuker = (override: boolean, cn: string, ecn: string): string => override ? cn : cx(ecn, cn);
