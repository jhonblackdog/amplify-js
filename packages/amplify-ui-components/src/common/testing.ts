export const pixelThreshold = 10;
