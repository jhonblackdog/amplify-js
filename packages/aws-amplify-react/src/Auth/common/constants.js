export const Constants = {
    AUTH_SOURCE_KEY: 'amplify-react-auth-source',
    AUTH0: 'auth0',
    GOOGLE: 'google',
    FACEBOOK: 'facebook',
    AMAZON: 'amazon',
    SIGN_IN_WITH_HOSTEDUI_KEY: 'amplify-react-signin-with-hostedUI'
};
