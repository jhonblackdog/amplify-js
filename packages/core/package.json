{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
