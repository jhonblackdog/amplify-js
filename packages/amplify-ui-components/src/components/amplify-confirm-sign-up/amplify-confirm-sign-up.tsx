import { I18n } from '@aws-amplify/core';
import { Component, Prop, h, State } from '@stencil/core';
import { FormFieldTypes } from '../amplify-auth-fields/amplify-auth-fields-interface';
import { NO_AUTH_MODULE_FOUND } from '../../common/constants';
import { Translations } from '../../common/Translations';
import { AuthState, CognitoUserInterface, AuthStateHandler, UsernameAliasStrings } from '../../common/types/auth-types';

import { Auth } from '@aws-amplify/auth';
import { dispatchToastHubEvent, dispatchAuthStateChangeEvent, checkUsernameAlias } from '../../common/helpers';

@Component({
  tag: 'amplify-confirm-sign-up',
  shadow: true,
})
export class AmplifyConfirmSignUp {
  /** Fires when sign up form is submitted */
  @Prop() handleSubmit: (submitEvent: Event) => void = event => this.confirmSignUp(event);
  /** Used for header text in confirm sign up component */
  @Prop() headerText: string = I18n.get(Translations.CONFIRM_SIGN_UP_HEADER_TEXT);
  /** Used for the submit button text in confirm sign up component */
  @Prop() submitButtonText: string = I18n.get(Translations.CONFIRM_SIGN_UP_SUBMIT_BUTTON_TEXT);
  /**
   * Form fields allows you to utilize our pre-built components such as username field, code field, password field, email field, etc.
   * by passing an array of strings that you would like the order of the form to be in. If you need more customization, such as changing
   * text for a label or adjust a placeholder, you can follow the structure below in order to do just that.
   * ```
   * [
   *  {
   *    type: string,
   *    label: string,
   *    placeholder: string,
   *    hint: string | Functional Component | null,
   *    required: boolean
   *  }
   * ]
   * ```
   */
  @Prop() formFields: FormFieldTypes | string[] = [];
  /** Auth state change handler for this components
   * e.g. SignIn -> 'Create Account' link -> SignUp
   */
  @Prop() handleAuthStateChange: AuthStateHandler = dispatchAuthStateChangeEvent;
  /** Used for the username to be passed to resend code */
  @Prop() user: CognitoUserInterface;
  /** Username Alias is used to setup authentication with `username`, `email` or `phone_number`  */
  @Prop() usernameAlias: UsernameAliasStrings = 'username';

  @State() code: string;
  @State() loading: boolean = false;
  @State() userInput: string = this.user ? this.user.username : null;
  private _signUpAttrs = this.user && this.user.signUpAttrs ? this.user.signUpAttrs : null;
  private newFormFields: FormFieldTypes | string[] = [];

  componentWillLoad() {
    checkUsernameAlias(this.usernameAlias);
    if (this.formFields.length === 0) {
      this.newFormFields = [
        {
          type: `${this.usernameAlias}`,
          required: true,
          handleInputChange: this.handleFormFieldInputChange(`${this.usernameAlias}`),
          value: this.userInput,
          disabled: this.userInput ? true : false,
        },
        {
          type: 'code',
          label: I18n.get(Translations.CONFIRM_SIGN_UP_CODE_LABEL),
          placeholder: I18n.get(Translations.CONFIRM_SIGN_UP_CODE_PLACEHOLDER),
          required: true,
          hint: (
            <div>
              {I18n.get(Translations.CONFIRM_SIGN_UP_LOST_CODE)}{' '}
              <amplify-button variant="anchor" onClick={() => this.resendConfirmCode()}>
                {I18n.get(Translations.CONFIRM_SIGN_UP_RESEND_CODE)}
              </amplify-button>
            </div>
          ),
          handleInputChange: this.handleFormFieldInputChange('code'),
        },
      ];
    } else {
      this.formFields.forEach(field => {
        const newField = { ...field };
        newField['handleInputChange'] = event => this.handleFormFieldInputWithCallback(event, field);
        this.newFormFields.push(newField);
      });
    }
  }

  handleFormFieldInputChange(fieldType) {
    switch (fieldType) {
      case 'username':
      case 'email':
      case 'phone_number':
        return event => (this.userInput = event.target.value);
      case 'code':
        return event => (this.code = event.target.value);
      default:
        return;
    }
  }

  handleFormFieldInputWithCallback(event, field) {
    const fnToCall = field['handleInputChange']
      ? field['handleInputChange']
      : (event, cb) => {
          cb(event);
        };
    const callback = this.handleFormFieldInputChange(field.type);
    fnToCall(event, callback.bind(this));
  }

  async resendConfirmCode() {
    if (event) {
      event.preventDefault();
    }
    if (!Auth || typeof Auth.resendSignUp !== 'function') {
      throw new Error(NO_AUTH_MODULE_FOUND);
    }
    try {
      if (!this.userInput) throw new Error('Username can not be empty');
      await Auth.resendSignUp(this.userInput);
      this.handleAuthStateChange(AuthState.ConfirmSignUp);
    } catch (error) {
      dispatchToastHubEvent(error);
    }
  }

  // TODO: Add validation
  // TODO: Prefix
  async confirmSignUp(event: Event) {
    if (event) {
      event.preventDefault();
    }
    if (!Auth || typeof Auth.confirmSignUp !== 'function') {
      throw new Error(NO_AUTH_MODULE_FOUND);
    }

    this.loading = true;

    try {
      const confirmSignUpResult = await Auth.confirmSignUp(this.userInput, this.code);
      const user =
        confirmSignUpResult && this._signUpAttrs && (await Auth.signIn(this.userInput, this._signUpAttrs.password));
      this.handleAuthStateChange(AuthState.SignedIn, user);
    } catch (error) {
      dispatchToastHubEvent(error);
    } finally {
      this.loading = false;
    }
  }

  render() {
    return (
      <amplify-form-section
        headerText={this.headerText}
        submitButtonText={this.submitButtonText}
        handleSubmit={this.handleSubmit}
        secondaryFooterContent={
          <div>
            <span>
              <amplify-button variant="anchor" onClick={() => this.handleAuthStateChange(AuthState.SignIn)}>
                {I18n.get(Translations.BACK_TO_SIGN_IN)}
              </amplify-button>
            </span>
          </div>
        }
      >
        <amplify-auth-fields formFields={this.newFormFields} />
      </amplify-form-section>
    );
  }
}
