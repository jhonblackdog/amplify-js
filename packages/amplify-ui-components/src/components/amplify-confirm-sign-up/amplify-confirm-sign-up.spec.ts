import { newSpecPage } from '@stencil/core/testing';
import { AmplifyConfirmSignUp } from './amplify-confirm-sign-up';
import { CONFIRM_SIGN_UP_HEADER_TEXT, CONFIRM_SIGN_UP_SUBMIT_BUTTON_TEXT } from '../../common/constants';

describe('amplify-confirm-sign-up spec:', () => {
  describe('Component logic ->', () => {
    let confirmSignUp;

    beforeEach(() => {
      confirmSignUp = new AmplifyConfirmSignUp();
    });

    it('should render `handleSubmit` as defined by default', () => {
      expect(confirmSignUp.handleSubmit).toBeDefined();
    });

    it('should render `validationErrors` as undefined by default', () => {
      expect(confirmSignUp.validationErrors).toBeUndefined();
    });

    it('should render `headerText` to `Confirm Sign up` by default', () => {
      expect(confirmSignUp.headerText).toEqual(CONFIRM_SIGN_UP_HEADER_TEXT);
    });

    it('should render `submitButtonText` to `Confirm` by default', () => {
      expect(confirmSignUp.submitButtonText).toEqual(CONFIRM_SIGN_UP_SUBMIT_BUTTON_TEXT);
    });

    it('should render `overrideStyle` to false by default', () => {
      expect(confirmSignUp.overrideStyle).toBe(false);
    });
  });
  describe('Render logic ->', () => {
    it('should render a `confirm sign up` form by default', async () => {
      const page = await newSpecPage({
        components: [AmplifyConfirmSignUp],
        html: `<amplify-confirm-sign-up></amplify-confirm-sign-up>`,
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});
