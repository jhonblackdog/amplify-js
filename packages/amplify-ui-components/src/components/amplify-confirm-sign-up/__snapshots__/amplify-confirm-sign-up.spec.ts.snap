// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-confirm-sign-up spec: Render logic -> should render a \`confirm sign up\` form by default 1`] = `
<amplify-confirm-sign-up>
  <amplify-form-section headertext="Confirm Sign up" submitbuttontext="Confirm">
    <amplify-auth-fields></amplify-auth-fields>
  </amplify-form-section>
</amplify-confirm-sign-up>
`;
